tls/s2n_handshake.h

```diff
@@ -77,6 +78,9 @@ struct s2n_handshake {
 /* Handshake needs OCSP status message */
 #define OCSP_STATUS                 0x08
 
+/* Session Resumption via session-tickets */
+#define WITH_SESSION_TICKET         0x20
```

Comment:
I was trying to foreshadow the eventual merge with mutual-auth (which uses 0x10) (hence the 0x20 and the 64 length for the state-machine datastructure).  I can switch back to 0x10 if we want to treat them independently and worry about the clash later during merging later